### ISSUE TITLE
Get consent before setting cookies

### ISSUE THREAD

**Issue by ghost** [backdrop-contrib/eu_cookie_compliance#5]
It sounds like this module adds a banner telling users about the use of cookies on the site, but from my understanding you need their active consent before setting any cookies. Does this module handle that? If not, can/should it?

Related: https://github.com/backdrop/backdrop-issues/issues/3094

**Comment by Graham-72** [comment]
This module is a port from the Drupal module but has not been updated for a while. The D7 version has recently been updated to 7.x-1.25 and its description at https://www.drupal.org/project/eu_cookie_compliance suggests that it can be used to prevent cookies being set on the site.

Is this module sufficiently useful to Backdrop to make it worthwhile spending the effort to update this current port? Are there other possibilities? We do need whatever is necessary to fulfil GDPR requirements, if these can be substantiated. (There is a lot of hype about!) I look forward to some answers from Issue #3094.

**Comment by Graham-72** [comment]
The presentation given at Twin Cities Drupal 'The GDPR is here' https://www.youtube.com/watch?v=CyIFNsSHPxQ suggests that this module is not satisfactory for meeting the new EU requirements and that something more is needed, e.g. the blizz_vanisher module, so I am having a look at that https://www.drupal.org/project/blizz_vanisher.

The stumbling block seems to be that one must now prevent cookies being set for any external content being included in one's site, unless the user gives active consent.

**Comment by sbgraphicdesign** [comment]
Hi Graham, how did you get on with Blizz Vanisher? I tried a straight port from the D7 version, but am getting errors (related to the entity module) when it's enabled. I'm no programmer/coder so I've no idea how to solve this.

However I feel it's imperative to have some form of cookie module that is GDPR ready in order to use Backdrop in the EU (unless, of course, you have no tracking cookies in your site!).

**Comment by Graham-72** [comment]
Note for the record: for Blizz Vanisher see the Backdrop CMS module https://github.com/backdrop-contrib/gdpr_cookies.

**Comment by swampopus** [comment]
@Graham-72 I'm curious if this is still an issue.  According to the commit log, "Working Backdrop conversion of 7.x-1.26" that makes it sound like this IS fixed, and the module does correctly prevent cookies from being set.

I know gdpr_cookies is being actively maintained, but this module is much prettier, and I'd prefer to use it if I can.  If it DOES correctly block cookies from being set, I think updating the README.md file to say so would be a good idea.

**Comment by nattywebdev** [comment]
I agree with the comment by @swampopus - would prefer to use the eu_cookie_compliance module but if it doesn't actually stop cookies from being set, it's really not much use!

**Comment by swampopus** [comment]
@nattywebdev 
Okay-- look at the JS likes starting at   https://github.com/backdrop-contrib/eu_cookie_compliance/blob/2c79ceb482092e7e4de881e342109b0eefebc582/js/eu_cookie_compliance.js#L437

Also, the comment on line 448: https://github.com/backdrop-contrib/eu_cookie_compliance/blob/2c79ceb482092e7e4de881e342109b0eefebc582/js/eu_cookie_compliance.js#L448

It definitely looks like it blocks or loads cookies based on a whitelist.  But, the best I can tell, this is something you manually have to edit in the settings json file: https://github.com/backdrop-contrib/eu_cookie_compliance/blob/2c79ceb482092e7e4de881e342109b0eefebc582/config/eu_cookie_compliance.settings.json#L11

At least, I couldn't find anywhere to enter something through the web UI config page.  I feel like we just need some documentation. Ex: common entries you should enter for things like Google Analytics.

That's assuming I'm right about the whitelist config setting.

**Comment by swampopus** [comment]
You know, now that I'm looking at it some more, I think I'm wrong about the whitelisted_cookies setting.  I think it's just used for essential cookies that *must* be accepted.

And then, starting on line 466: https://github.com/backdrop-contrib/eu_cookie_compliance/blob/2c79ceb482092e7e4de881e342109b0eefebc582/js/eu_cookie_compliance.js#L466
I think it just removes _all_ cookies that aren't explicitly whitelisted.

So... I think this is good... I think it is working the way we hope.  I will point out that it's using jquery.cookie, which is no longer supported (hasn't been updated in 10 years), and recommends instead switching to js-cookie.  But, one thing at a time :/
